### PR TITLE
Node 14 changes

### PR DESCRIPTION
**Pull request by toddhgardner**
Changes to the agent to support Node 14 LTS upgrade on our build servers:

- Jest behavior when mocking console changed, needed to switch to the `spyOn` syntax
- Domain behavior changed and no longer remembers the active domain between async express functions. Added a patch onto the `request` object to maintain our reference to it.
- Prep-work for Node 15, where the promise rejection issue within node will be resolved. Node 15 will restrict `defineProperty` to throw when used on a string, which was possible in promise rejections. Added logic to Agent to force non-errors into Error instances.

### CODE REVIEW

File: src/Agent.ts [hunk 1]
```diff
@@ -65,6 +65,9 @@ export class Agent {
     if (error["__trackjs__"]) {
       return false;
     }
+    if (!(error as any instanceof Error)) {
+      error = new Error("" + error);
+    }
```
Comment (toddhgardner) [inline src/Agent.ts:70]: We might want to considering porting our full-featured serialize function over from the browser agent, but that felt like overkill for now. There are less bad-practices around errors in the Node ecosystem 🤞

File: src/handlers/express.ts [hunk 2]
```diff
@@ -36,15 +38,18 @@ export function expressRequestHandler(
...
 
     // execute the remaining middleware within the context of this domain.
     requestDomain.run(() => {
       let agent = AgentRegistrar.getCurrentAgent();
+      req[DOMAIN_CARRIER_KEY] = requestDomain;
       let correlationId = uuid();
 
       // Configure the agent to handle the details of this request.
-      agent.configure({ correlationId: uuid() }); // correlate all errors from this request together.
+      agent.configure({ correlationId: correlationId }); // correlate all errors from this request together.
```
Comment (toddhgardner) [inline src/handlers/express.ts:52]: This was just a straight-up bug. Oops.

**Comment by BrandesEric** [comment]
Looks like great changes to me!  Thanks for tracking all this down!

**Comment by toddhgardner** [comment]
I'm going to send this to canary, and get it pulled in on RemoteJS for bake for a bit.

**Pull request merged**